src/MacroTools/Frames/Books/ArtifactSystem/ArtifactBook.cs

```diff
@@ -51,12 +49,31 @@ private void OnArtifactDisposed(object? sender, Artifact artifact)
     {
       _pagesByArtifact[artifact].RemoveArtifact(artifact);
       _pagesByArtifact.Remove(artifact);
+      FillGaps();
+    }
+
+    private void FillGaps()
```

Comment:
This isn't just filling gaps, it's re-rendering the whole thing. Maybe we can just call it ReRender?

Reply:
Good idea. I wasn't happy with the Name either but ReRender sounds good